pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -475,6 +476,18 @@ func (ctrl *Controller) syncKubeletConfig(key string) error {
 		klog.V(4).Infof("Finished syncing kubeletconfig %q (%v)", key, time.Since(startTime))
 	}()
 
+	if err := wait.PollUntilContextTimeout(context.TODO(), ctrlcommon.ControllerConfigRolloutInterval, ctrlcommon.ControllerConfigTimeout, false, func(_ context.Context) (bool, error) {
+		_, err := ctrl.client.MachineconfigurationV1().ControllerConfigs().Get(context.TODO(), ctrlcommon.ControllerConfigName, metav1.GetOptions{})
+		if err != nil {
```

Comment:
suggestion (non-blocking): I tend to follow this error-checking convention whenever I'm looking for a specific error:

```golang
// Start with the happy path.
if err == nil {
	return true, nil 
}

// Next, check for specific errors that I want to mask or otherwise handle here.
if apierrors.IsNotFound(err) {
	return false, nil
}

// Return all other errors to the caller.
return false, err
```

This avoids a lot of nested if statements and unnecessary indention while simultaneously being easier to mentally parse.